src/components/dropdown/daikin-dropdown.ts

```diff
@@ -0,0 +1,426 @@
+import {
+  flip,
+  offset,
+  size,
+  type ComputePositionConfig,
+} from "@floating-ui/dom";
+import { cva } from "class-variance-authority";
+import { LitElement, css, html, unsafeCSS, type PropertyValues } from "lit";
+import {
+  customElement,
+  property,
+  query,
+  queryAssignedElements,
+  state,
+} from "lit/decorators.js";
+import { ClickOutsideController } from "../../controllers/click-outside";
+import { FloatingUIAutoUpdateController } from "../../controllers/floating-ui-auto-update";
+import tailwindStyles from "../../tailwind.css?inline";
+import type { DaikinDropdownItem } from "../dropdown-item";
+import type { DaikinInputGroup } from "../input-group";
+
+const cvaButton = cva(
+  [
+    "flex",
+    "items-center",
+    "gap-2",
+    "w-full",
+    "h-full",
+    "min-h-12",
+    "bg-white",
+    "pl-4",
+    "pr-3",
+    "border",
+    "border-[--color-base]",
+    "rounded-md",
+    "overflow-hidden",
+    "font-daikinSerif",
+    "text-left",
+    "leading-5",
+    "relative",
+    "outline",
+    "outline-0",
+    "-outline-offset-2",
+
+    "focus-visible:outline-2",
+    "focus-visible:outline-[--color-focus]",
+
+    "disabled:text-daikinNeutral-200",
+    "disabled:border-daikinNeutral-200",
+
+    "after:i-daikin-dropdown-chevron-down",
+    "after:w-6",
+    "after:h-6",
+    "after:absolute",
+    "after:m-auto",
+    "after:top-0",
+    "after:bottom-0",
+    "after:right-3",
+  ],
+  {
+    variants: {
+      open: {
+        false: [
+          "enabled:hover:bg-[#f2f2f2]",
+          "enabled:active:bg-daikinNeutral-100",
+        ],
+        true: ["enabled:bg-daikinNeutral-100"],
+      },
+      error: {
+        false: [
+          "var-color-daikinNeutral-600/color-base",
+          "var-color-daikinBlue-700/color-focus",
+        ],
+        true: [
+          "var-color-daikinRed-500/color-base",
+          "var-color-daikinRed-500/color-focus",
+        ],
+      },
+      placeholder: {
+        false: ["text-daikinNeutral-900"],
+        true: ["text-daikinNeutral-700"],
+      },
+    },
+  }
+);
+
+const floatingPositionOptions: Partial<ComputePositionConfig> = {
+  placement: "bottom",
+  middleware: [
+    flip({ fallbackStrategy: "initialPlacement" }),
+    offset({ mainAxis: 0 }),
+    size({
+      apply({ rects, elements }) {
+        Object.assign(elements.floating.style, {
+          minWidth: `${rects.reference.width}px`,
+        });
```

Comment:
I think using CSS variables is better than directly applying styles.

```suggestion
        elements.floating.style.setProperty("--floating-width", `${rects.reference.width}px`);
```